src/Profilers/MutatingMonitorProfiler/ProbeInstrumentation/ProbeInstrumentation.cpp

```diff
@@ -47,9 +63,84 @@ HRESULT ProbeInstrumentation::RegisterFunctionProbe(FunctionID enterProbeId)
 HRESULT ProbeInstrumentation::InitBackgroundService()
 {
     m_probeManagementThread = thread(&ProbeInstrumentation::WorkerThread, this);
+    //
+    // Create a dedicated thread for managed callbacks.
+    // Performing the callbacks will taint the calling thread preventing it
+    // from using certain ICorProfiler APIs marked as unsafe.
+    // Those functions will fail with CORPROF_E_UNSUPPORTED_CALL_SEQUENCE.
+    //
+    m_managedCallbackThread = thread(&ProbeInstrumentation::ManagedCallbackThread, this);
```

Comment:
nit: Is there a way we can name both of these threads so we can identify them under the debugger?